src/npctalk.cpp

```diff
@@ -2316,6 +2316,39 @@ void talk_effect_fun_t::set_queue_effect_on_condition( const JsonObject &jo,
         }
     };
 }
+void talk_effect_fun_t::set_add_morale( const JsonObject &jo, const std::string &member,
+                                        bool is_npc )
+{
+    std::string new_type = jo.get_string( member );
+    int bonus = jo.get_int( "bonus" );
+    int max_bonus = jo.get_int( "max_bonus", 0 );
```

Comment:
This has no default `bonus` and a default for `max_bonus` of 0?

Reply:
The defaults are the default values for the parameters of the morale add function on morale.h line 33.